scripts/cplane/kube_dashboard.sh

```diff
@@ -0,0 +1,197 @@
+
+    echo "⚙️  Apply recommended manifests from the upstream project:"
+    kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.7.0/aio/deploy/recommended.yaml
+}
+
+
+# Create RBAC credential to access / admin via dashboard
+function dashboard_user_create {
+    echo "⚙️  Create user credentials for dashboard"
+    echo "⚙️  First, YAML file to create dashboard admin user"
+
+    #
+    # This creates the yaml file to instruct k8s in creating the dashboard space and the rbac access
+    # Do NOT modify or change the spacing between these lines!
+    #
+    # ----------------------------------------------
+    cat > dashboard-adminuser.yaml << EOF
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: admin-user
+  namespace: ${NAMESPACE}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: admin-user-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: admin-user
+  namespace: ${NAMESPACE}
+EOF
+# ----------------------------------------------
+#
+    echo "⚙️  Apply user create yaml to k8s"
+    kubectl apply -f dashboard-adminuser.yaml
+    rm -rf dashboard-adminuser.yaml
+}
+
+
+# Retrieve and display the dashboard credentials
+# And the url for login
+function dashboard_user_token {
+    echo "🔍 Retrieving the dashboard login token from k8s"
+    echo "⚙️  Copy and paste this to the 'Enter token *' field on the dashboard login"
+    echo "--------------------------------------------------"
+    kubectl -n ${NAMESPACE} create token admin-user
+    echo "--------------------------------------------------"
+    echo "Dashboard is at: https://${CPLANE_IP}:${DASHBOARD_PORT}"
+    echo ""
+}
+
+
+# Specific the specific port for the Dashboard
+# NodePort will otherwise choose a random port
+function dashboard_port_forward {
+    echo "⚙️  Setting up dashboard access via NodePort"
+    kubectl -n ${NAMESPACE} patch svc ${SERVICE} -p '{"spec": {"type": "NodePort"}}'
+
+    echo "🔍 Check Assigned NodePort:"
+    kubectl -n ${NAMESPACE} get svc ${SERVICE}
+
+    echo "⚙️  Patching service '${SERVICE}' to use nodePort=${DASHBOARD_PORT}..."
+    kubectl -n "${NAMESPACE}" patch svc "${SERVICE}" --type='json' -p="[
+        {
+            \"op\": \"replace\",
+            \"path\": \"/spec/ports/0/nodePort\",
+            \"value\": $((DASHBOARD_PORT))
+        },
+        {
+            \"op\": \"replace\",
+            \"path\": \"/spec/type\",
+            \"value\": \"NodePort\"
+        }
+    ]"
+
+    if [ $? -eq 0 ]; then
+        echo "✅ Dashboard is now exposed at: https://${CPLANE_IP}:${DASHBOARD_PORT}/"
+    else
+        echo "❌ Failed to patch the Service"
+        exit 1
+    fi
+
+    echo "🔍 New Assigned NodePort:"
+    kubectl -n ${NAMESPACE} get svc ${SERVICE}
+}
+
+
+#
+# Main Execution Loop
+#
+# Grab the command parameter
```

Comment:
param check. The ${var,,} forces the value to lower case, which makes for cleaner Bash comparisons.
One wants to take "ThIsStRiNg" to "thisstring" for comparison